File: priceoracle/ts/priceoracle/priceoracle.ts
// Copyright 2020 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import * as wasmlib from "wasmlib";
import * as wasmtypes from "wasmlib/wasmtypes";
import * as sc from "./index";

export function funcApproveOracle(ctx: wasmlib.ScFuncContext, f: sc.ApproveOracleContext): void {
}

export function funcInit(ctx: wasmlib.ScFuncContext, f: sc.InitContext): void {
	if (f.params.owner().exists()) {
		f.state.owner().setValue(f.params.owner().value());
		return;
	}
	f.state.owner().setValue(ctx.contractCreator());
}

export function funcSetMiotaPrice(ctx: wasmlib.ScFuncContext, f: sc.SetMiotaPriceContext): void {
}

export function funcSetOwner(ctx: wasmlib.ScFuncContext, f: sc.SetOwnerContext): void {
	f.state.owner().setValue(f.params.owner().value());
}

export function viewGetMiotaPrice(ctx: wasmlib.ScViewContext, f: sc.GetMiotaPriceContext): void {
}

export function viewGetOwner(ctx: wasmlib.ScViewContext, f: sc.GetOwnerContext): void {
	f.results.owner().setValue(f.state.owner().value());
}

File: priceoracle/go/priceoracle/priceoracle.go
// Copyright 2020 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

package priceoracle

import "github.com/iotaledger/wasp/packages/wasmvm/wasmlib/go/wasmlib"

func funcApproveOracle(ctx wasmlib.ScFuncContext, f *ApproveOracleContext) {
}

func funcInit(ctx wasmlib.ScFuncContext, f *InitContext) {
	if f.Params.Owner().Exists() {
		f.State.Owner().SetValue(f.Params.Owner().Value())
		return
	}
	f.State.Owner().SetValue(ctx.ContractCreator())
}

func funcSetOwner(ctx wasmlib.ScFuncContext, f *SetOwnerContext) {
	f.State.Owner().SetValue(f.Params.Owner().Value())
}

func viewGetOwner(ctx wasmlib.ScViewContext, f *GetOwnerContext) {
	f.Results.Owner().SetValue(f.State.Owner().Value())
}

func funcSetMiotaPrice(ctx wasmlib.ScFuncContext, f *SetMiotaPriceContext) {
}

func viewGetMiotaPrice(ctx wasmlib.ScViewContext, f *GetMiotaPriceContext) {
}
